pkg/emulator/main.go

```diff
@@ -0,0 +1,187 @@
+package main
```

Comment:
Since we can't have a `.mod` file, we also can't have a main here.
So we need this to be a test, as in `emulator_test.go` instead of `main.go`.
And package must be `emulator`.